lib/pbench/server/api/resources/query_apis/datasets_detail.py

```diff
@@ -0,0 +1,266 @@
+from logging import Logger
+
+from flask import request, jsonify
+from flask_restful import Resource, abort
+import requests
+
+from dateutil import parser
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources.query_apis import (
+    get_es_url,
+    get_index_prefix,
+    gen_month_range,
+    get_user_term,
+)
+
+
+class DatasetsDetail(Resource):
```

Comment:
Bummer, the state tracking uses the name dataset, a tar ball with metadata.  It is unfortunate that we are introducing a new name.  Could we consider renaming `Dataset` to `Tarball`?  Big can of worms, but I really feel "dataset" is too abstract.  It is really an opaque object that is known to be a tar ball, with metadata about it.